x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/three_way_diff/three_way_diff_outcome.ts

```diff
@@ -38,7 +38,55 @@ export const determineDiffOutcome = <TValue>(
   const baseEqlTarget = isEqual(baseVersion, targetVersion);
   const currentEqlTarget = isEqual(currentVersion, targetVersion);
 
-  if (baseVersion === MissingVersion) {
+  return getThreeWayDiffOutcome({
+    baseEqlCurrent,
+    baseEqlTarget,
+    currentEqlTarget,
+    hasBaseVersion: baseVersion !== MissingVersion,
+  });
+};
+
+/**
+ * Determines diff outcomes of array fields that do not care about order (e.g. `[1, 2 , 3] === [3, 2, 1]`)
+ */
+export const determineOrderAgnosticDiffOutcome = <TValue>(
+  baseVersion: TValue[] | MissingVersion,
+  currentVersion: TValue[],
+  targetVersion: TValue[]
+): ThreeWayDiffOutcome => {
+  let baseEqlCurrent: boolean;
+  let baseEqlTarget: boolean;
+  if (baseVersion !== MissingVersion) {
+    baseEqlCurrent = arraysHaveSameElements(baseVersion, currentVersion);
+    baseEqlTarget = arraysHaveSameElements(baseVersion, targetVersion);
+  } else {
+    baseEqlCurrent = false;
+    baseEqlTarget = false;
+  }
+  const currentEqlTarget = arraysHaveSameElements(currentVersion, targetVersion);
```

Comment:
`_.isEqual` should be able to compare sets between each other, as well as a set to a Symbol. So we could simplify this to:

```suggestion
  const baseSet = baseVersion === MissingVersion ? MissingVersion : new Set<TValue>(baseVersion);
  const currentSet = new Set<TValue>(currentVersion);
  const targetSet = new Set<TValue>(targetVersion);

  const baseEqlCurrent = isEqual(baseSet, currentSet);
  const baseEqlTarget = isEqual(baseSet, targetSet);
  const currentEqlTarget = isEqual(currentSet, targetSet);
```

And we won't need a `arraysHaveSameElements` or a `areSetsEqual` function.